modules/pixi/styles.js

```diff
@@ -30,568 +31,531 @@ import { osmPavedTags } from '../osm/tags';
-    stroke: { width: 3, color: 0x998888, dash: [6, 6], cap: PIXI.LINE_CAP.BUTT }
-  },
-  crossing_marked: {
-    casing: { width: 5, color: 0xddccaa },
-    stroke: { width: 3, color: 0x4c4444, dash: [6, 3], cap: PIXI.LINE_CAP.BUTT }
-  },
-  crossing_unmarked: {
-    casing: { width: 5, color: 0xddccaa },
-    stroke: { width: 3, color: 0x776a6a, dash: [6, 4], cap: PIXI.LINE_CAP.BUTT }
-  },
-  cycleway: {
-    casing: { width: 5, color: 0xffffff },
-    stroke: { width: 3, color: 0x58a9ed, dash: [6, 6], cap: PIXI.LINE_CAP.BUTT }
-  },
-  bridleway: {
-    casing: { width: 5, color: 0xffffff },
-    stroke: { width: 3, color: 0xe06d5f, dash: [6, 6], cap: PIXI.LINE_CAP.BUTT }
-  },
-  corridor: {
-    casing: { width: 5, color: 0xffffff },
-    stroke: { width: 3, color: 0x8cd05f, dash: [2, 8], cap: PIXI.LINE_CAP.ROUND }
-  },
-  steps: {
-    casing: { width: 5, color: 0xffffff },
-    stroke: { width: 3, color: 0x81d25c, dash: [3, 3], cap: PIXI.LINE_CAP.BUTT }
-  },
-
-  river: {
-    casing: { width: 10, color: 0x444444 },
-    stroke: { width: 8, color: 0x77dddd }
-  },
-  stream: {
-    casing: { width: 7, color: 0x444444 },
-    stroke: { width: 5, color: 0x77dddd }
-  },
-  ridge: {
-    stroke: { width: 2, color: 0x8cd05f}  // rgb(140, 208, 95)
-  },
-
-  runway: {
-    casing: { width: 10, color: 0x000000, cap: PIXI.LINE_CAP.BUTT },
-    stroke: { width: 8, color: 0xffffff, dash: [24, 48], cap: PIXI.LINE_CAP.BUTT }
-  },
-  taxiway: {
-    casing: { width: 7, color: 0x444444 },
-    stroke: { width: 5, color: 0xffff00 }
-  },
-
-  railway: {
-    casing: { width: 7, color: 0x555555, cap: PIXI.LINE_CAP.BUTT },
-    stroke: { width: 2, color: 0xeeeeee, dash: [12, 12], cap: PIXI.LINE_CAP.BUTT,  }
-  },
-
-  ferry: {
-    casing: { alpha: 0 },  // disable
-    stroke: { width: 3, color: 0x58a9ed, dash: [12, 8], cap: PIXI.LINE_CAP.BUTT }
-  },
-
-  boundary: {
-    casing: { width: 6, color: 0x82b5fe, cap: PIXI.LINE_CAP.BUTT },
-    stroke: { width: 2, color: 0xffffff, dash: [20, 5, 5, 5], cap: PIXI.LINE_CAP.BUTT }
-  },
-  boundary_park: {
-    casing: { width: 6, color: 0x82b5fe, cap: PIXI.LINE_CAP.BUTT },
-    stroke: { width: 2, color: 0xb0e298, dash: [20, 5, 5, 5], cap: PIXI.LINE_CAP.BUTT }
-  },
-
-  barrier: {
-    casing: { alpha: 0 },  // disable
-    stroke: { width: 3, color: 0xdddddd, dash: [10, 5, 1, 5], cap: PIXI.LINE_CAP.ROUND }
-  },
-  barrier_wall: {
-    casing: { alpha: 0 },  // disable
-    stroke: { width: 3, color: 0xdddddd, dash: [10, 5, 1, 5], cap: PIXI.LINE_CAP.ROUND }
-  },
-  barrier_hedge: {
-    fill:   { color: 0x8cd05f, alpha: 0.3 },   // rgb(140, 208, 95)
-    casing: { alpha: 0 },  // disable
-    stroke: { width: 3, color: 0x8cd05f, dash: [10, 5, 1, 5], cap: PIXI.LINE_CAP.ROUND }
-  },
-
-  tree_row: {
-    casing: { width: 7, color: 0x444444 },
-    stroke: { width: 5, color: 0x8cd05f }
+export class Styles {
+  constructor(context) {
+    this.context = context;
```

Comment:
@Bonkles in doing some debugging today, I realized that `context` here is referring to the WebGL canvas 🤔 

<img width="741" alt="Screenshot 2023-11-03 at 10 16 04 AM" src="https://github.com/facebook/Rapid/assets/95086147/969170ae-76da-423a-90df-e1a1038630b9">

Reply:
Yup depending on where you are in the code, context can refer to different things.  WebGL Canvas has a context, and then there's the 'Rapid' specific Context. You can think of it like this: 

```
WebGL/Pixi Context { 
let context = {};

  Rapid { 
    // same variable name that's 'masking' the variable in the calling scope
    let context = {}; 
    
  }

}
```

In this case we want to be handing the rapid context into the Style class, so we should examine where that's coming from. 

This is why re-using variable names is discouraged, but is sometimes unavoidable when you combine different libraries together!

Reply:
I say this with lots of ⚠️ caution ⚠️  - but I got the styles class to work without errors (eek) and grab the color data from the `ColorSystem` `getColorScheme` class method. I still need to work on fixing `getColorScheme` so it doesn't return a static json object, which is my next task. Would you mind taking a look at the styles class for now?

Reply:
Just looked over the code - I don't see how static functions would work if they need access to `context` in order to find the ColorSystem.  The `context` gets established in the constructor.

Maybe the `ColorSystem` should really be a `StyleSystem` that is responsible for colors and styling?

Reply:
@bhousel I think you're onto something here. I mentioned that we might want to make the Styles class into a full StyleSystem but I felt that was a bit too feature-creepy for her at the moment, so I suggested just making it into a class with a single static method (for now).

Reply:
And @bhousel you're right- in order for a static method to work properly by using context, there'd have to be an instance of the class constructed with that context. I don't know what I was thinking yesterday...